WordPress/Classes/ViewRelated/NUX/SignupUsernameTableViewController.swift

```diff
@@ -0,0 +1,237 @@
+        switch section {
+        case Sections.titleAndDescription.rawValue:
+            return 1
+        case Sections.searchField.rawValue:
+            return suggestions.count > 0 ? 1 : 0
+        case Sections.suggestions.rawValue:
+            return suggestions.count
+        default:
+            return 0
+        }
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell: UITableViewCell
+        switch indexPath.section {
+        case Sections.titleAndDescription.rawValue:
+            cell = titleAndDescriptionCell()
+        case Sections.searchField.rawValue:
+            cell = searchFieldCell()
+        case Sections.suggestions.rawValue:
+            fallthrough
+        default:
+            let suggestion = suggestions[indexPath.row]
+            cell = suggestionCell(domain: suggestion)
+        }
+        return cell
+    }
+
+    override func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        if section == Sections.suggestions.rawValue {
+            let footer = UIView()
+            footer.backgroundColor = WPStyleGuide.greyLighten20()
+            return footer
+        }
+        return nil
+    }
+
+    override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        if section == Sections.suggestions.rawValue {
+            return 0.5
+        }
+        return 0
+    }
+
+    // MARK: table view cells
+
+    private func titleAndDescriptionCell() -> UITableViewCell {
+        let description = String(format: NSLocalizedString("Your username is currently \"%@\". It will be used for mentions and links, but otherwise people will just see your display name, \"%@\"", comment: "Description of how to pick a domain name during the site creation process"), currentUsername ?? "", displayName ?? "")
+        let cell = LoginSocialErrorCell(title: "", description: description)
+        cell.selectionStyle = .none
+        return cell
+    }
+
+    private func searchFieldCell() -> SiteCreationDomainSearchTableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: SiteCreationDomainSearchTableViewCell.cellIdentifier) as? SiteCreationDomainSearchTableViewCell else {
+            return SiteCreationDomainSearchTableViewCell(placeholder: "")
+        }
+        cell.delegate = self
+        cell.selectionStyle = .none
+        return cell
+    }
+
+    private func suggestionCell(domain: String) -> UITableViewCell {
+        let cell = UITableViewCell()
+
+        cell.textLabel?.attributedText = styleDomain(domain)
+        cell.textLabel?.textColor = WPStyleGuide.darkGrey()
+        cell.indentationWidth = SuggestionStyles.indentationWidth
+        cell.indentationLevel = SuggestionStyles.indentationLevel
+        return cell
+    }
+
+    private func styleDomain(_ domain: String) -> NSAttributedString {
```

Comment:
I don't think this is relevant for this `suggestionCell`.

Reply:
BUT I WANT TO KEEP IT AND HOLD IT